app/src/main/java/com/lagradost/cloudstream3/utils/DataStoreHelper.kt

```diff
@@ -352,20 +352,35 @@ object DataStoreHelper {
     /**
      * Used to display notifications on new episodes and posters in library.
      **/
-    data class SubscribedData(
+    abstract class LibrarySearchResponse(
         @JsonProperty("id") override var id: Int?,
-        @JsonProperty("subscribedTime") val bookmarkedTime: Long,
-        @JsonProperty("latestUpdatedTime") val latestUpdatedTime: Long,
-        @JsonProperty("lastSeenEpisodeCount") val lastSeenEpisodeCount: Map<DubStatus, Int?>,
+        @JsonProperty("latestUpdatedTime") open val latestUpdatedTime: Long,
         @JsonProperty("name") override val name: String,
         @JsonProperty("url") override val url: String,
         @JsonProperty("apiName") override val apiName: String,
-        @JsonProperty("type") override var type: TvType? = null,
+        @JsonProperty("type") override var type: TvType?,
         @JsonProperty("posterUrl") override var posterUrl: String?,
-        @JsonProperty("year") val year: Int?,
-        @JsonProperty("quality") override var quality: SearchQuality? = null,
-        @JsonProperty("posterHeaders") override var posterHeaders: Map<String, String>? = null,
-    ) : SearchResponse {
+        @JsonProperty("year") open val year: Int?,
+        @JsonProperty("imdbId") open val imdbId: String?,
+        @JsonProperty("quality") override var quality: SearchQuality?,
+        @JsonProperty("posterHeaders") override var posterHeaders: Map<String, String>?
+    ) : SearchResponse
+
+    data class SubscribedData(
+        @JsonProperty("subscribedTime") val subscribedTime: Long,
+        @JsonProperty("lastSeenEpisodeCount") val lastSeenEpisodeCount: Map<DubStatus, Int?>,
+        override var id: Int?,
+        override val latestUpdatedTime: Long,
+        override val name: String,
+        override val url: String,
+        override val apiName: String,
+        override var type: TvType?,
+        override var posterUrl: String?,
+        override val year: Int?,
+        override val imdbId: String?,
```

Comment:
If you are always passing something to it doesn't it become null from what you are using if it does not exist? The value is always passed so I thought it would work, might be wrong as the whole json thing kinda confuses me, so I can do that if needed.